src/testdrive/src/action/sql.rs

```diff
@@ -161,6 +161,26 @@ impl Action for SqlAction {
 }
 
 impl SqlAction {
+    /// Test data is inserted into a dataflow at some time t0. When we PEEK for the data,
+    /// there is no guarantee that the timestamp chosen for the PEEK will be t>t0. This uncertainty
+    /// causes us to query with a backoff for some amount of time, hoping that eventually the
+    /// time chosen for the PEEK will be t>t0.
+    ///
+    /// Instead of querying repeatedly, we can modify SELECT statements without an explicit
+    /// AS OF argument to always query with the max time (u64::max_value()), ensuring that we
+    /// should read all available data with the first query.
+    fn get_query_as_of(&self) -> Result<String, String> {
+        if let Statement::Select { query: _, as_of } = &self.stmt {
+            if as_of.is_none() {
+                // u64::max_value() is currently equivalent to dataflow-types' Timestamp::max_value
+                // without requiring the import.
```

Comment:
I don't know what our philosophy is here, but the `max_value()` function exists for a reason, which is that if anything surprising changes (for example, we change to using `isize` timestamps) that code changes too. I think if we ever change to `isize` timestamps this will all silently fail / hang, right?

Reply:
Yes, it would! I'm not using `u64::max_value()` here anymore